src/vsr/journal.zig

```diff
@@ -766,11 +783,21 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
                 }
             }
 
-            const read = journal.reads.acquire() orelse {
-                journal.read_prepare_log(op, checksum, "waiting for IOP");
-                callback(replica, null, null);
-                return;
-            };
+            if (destination_replica == null) {
+                journal.reads_commit_count += 1;
+            } else {
+                if (journal.reads_repair_count == journal.reads_repair_count_max) {
+                    journal.read_prepare_log(op, checksum, "waiting for IOP");
+                    callback(replica, null, null);
+                    return;
+                }
+                journal.reads_repair_count += 1;
+            }
+
+            assert(journal.reads_repair_count <= journal.reads_repair_count_max);
+            assert(journal.reads_commit_count <= journal.reads_commit_count_max);
+
+            const read = journal.reads.acquire() orelse unreachable;
```

Comment:
Is `orelse unreachable` instead of `acquire().?` to draw attention to its unreachable-ness?

Reply:
Heh, I actually wrote `.?` originally, but then it occured to me to check if there are other places where we branch on read IOP availability (as it seems nice to eliminate those), and all those places used `orelse unreachable`. But one wirte is using `.?`, and we use `.?` elsewhere, so that's what I am normalizing this to

Reply:
`orelse unreachable` was probably past-me not remembering that `.?` exists. :sweat_smile: